ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -54,12 +54,12 @@ Result conv2d(
     const std::optional<const Conv2dConfig>& conv_config_,
     const std::optional<const DeviceComputeKernelConfig>& compute_config_,
     const std::optional<const MemoryConfig>& memory_config) {
-    const bool mm_conv = use_matmul_for_1x1_conv(kernel_size, stride, padding, dilation, groups);
+    Conv2dConfig conv_config = conv_config_.value_or(Conv2dConfig());
+    const bool mm_conv = use_matmul_for_1x1_conv(kernel_size, stride, padding, dilation, groups) && !(conv_config.shard_layout.has_value() && conv_config.shard_layout.value() == TensorMemoryLayout::WIDTH_SHARDED);
```

Comment:
we use this in few palces (prepare weights transposed conv.. ), aren't those place going to have the same issue as this one?
Woudn't it be better to push additional logic into the function which checks if mm can be used?

Reply:
Will do

Reply:
Done.